flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -313,8 +318,6 @@ private void stopDispatcherServices() throws Exception {
 
 		final CompletableFuture<JobManagerRunner> jobManagerRunnerFuture = createJobManagerRunner(jobGraph);
 
-		jobManagerRunnerFutures.put(jobGraph.getJobID(), jobManagerRunnerFuture);
```

Comment:
Why this changes? Please revert if it is irrelevant. Separated futures/bugfixes and code cleans can help review.

Reply:
My changes probably changed timing for events happening in some unit tests and triggered a concurrency issue fixed by this change. The PR would have test instability without this change. I think it is small enough to be part of this PR, though, I can factor it out into a separate commit.